plugins/outputs/zabbix/zabbix_test.go

```diff
@@ -0,0 +1,845 @@
+				testutil.MustMetric("name",
+					map[string]string{
+						"host": "hostname",
+					},
+					map[string]interface{}{
+						"value": int64(0),
+					},
+					time.Unix(1522082244, 0),
+				),
+			},
+			zabbixMetrics: []zabbixRequestData{
+				{
+					Host:  "hostname",
+					Key:   "value",
+					Value: "0",
+					Clock: 1522082244,
+				},
+			},
+		},
+		"if AgentActive is configured, zabbix metrics should be sent respecting that protocol": {
+			AgentActive: true,
+			telegrafMetrics: []telegraf.Metric{
+				testutil.MustMetric("name",
+					map[string]string{
+						"host": "hostname",
+					},
+					map[string]interface{}{
+						"value": int64(0),
+					},
+					time.Unix(1522082244, 0),
+				),
+			},
+			zabbixMetrics: []zabbixRequestData{
+				{
+					Host:  "hostname",
+					Key:   "name.value",
+					Value: "0",
+					Clock: 1522082244,
+				},
+			},
+		},
+		"metrics should be time sorted, oldest to newest, to avoid zabbix doing extra work when generating trends": {
+			telegrafMetrics: []telegraf.Metric{
+				testutil.MustMetric("name",
+					map[string]string{
+						"host": "hostnameD",
+					},
+					map[string]interface{}{
+						"value": int64(0),
+					},
+					time.Unix(4444444444, 0),
+				),
+				testutil.MustMetric("name",
+					map[string]string{
+						"host": "hostnameC",
+					},
+					map[string]interface{}{
+						"value": int64(0),
+					},
+					time.Unix(3333333333, 0),
+				),
+				testutil.MustMetric("name",
+					map[string]string{
+						"host": "hostnameA",
+					},
+					map[string]interface{}{
+						"value": int64(0),
+					},
+					time.Unix(1111111111, 0),
+				),
+				testutil.MustMetric("name",
+					map[string]string{
+						"host": "hostnameB",
+					},
+					map[string]interface{}{
+						"value": int64(0),
+					},
+					time.Unix(2222222222, 0),
+				),
+			},
+			zabbixMetrics: []zabbixRequestData{
+				{
+					Host:  "hostnameA",
+					Key:   "name.value",
+					Value: "0",
+					Clock: 1111111111,
+				},
+				{
+					Host:  "hostnameB",
+					Key:   "name.value",
+					Value: "0",
+					Clock: 2222222222,
+				},
+				{
+					Host:  "hostnameC",
+					Key:   "name.value",
+					Value: "0",
+					Clock: 3333333333,
+				},
+				{
+					Host:  "hostnameD",
+					Key:   "name.value",
+					Value: "0",
+					Clock: 4444444444,
+				},
+			},
+		},
+	}
+
+	for desc, test := range tests {
+		t.Run(desc, func(t *testing.T) {
+			z := &Zabbix{
+				Address:               "127.0.0.1:10051",
+				Prefix:                test.Prefix,
+				HostTag:               "host",
+				SkipMeasurementPrefix: test.SkipMeasurementPrefix,
+				AgentActive:           test.AgentActive,
+				LLDSendInterval:       config.Duration(10 * time.Minute),
+				Log:                   testutil.Logger{},
+			}
+			require.NoError(t, z.Init())
+
+			// Simulate a Zabbix server to get the data sent. It has a timeout to avoid waiting forever.
+			listener, lerr := net.Listen("tcp", z.Address)
```

Comment:
I think there are special methods for creating listeners in tests without having fixed ports..

Reply:
fixed in 62968388e